### ISSUE TITLE
Can't see recent files

### ISSUE THREAD

**Issue by Scatchell** [athensresearch/athens#814]
**Problem**
When I use the "Find or Create a Page" (cmd + K) window, at the bottom is says: "Recent files". I assume there is meant to be a list of recent files here, but it is not populated for me. When I press the down or up arrow key it displays the error:

```
message=Uncaught TypeError: Cannot read property 'getBoundingClientRect' of null
source=file:///Applications/Athens.app/Contents/Resources/app.asar/resources/public/js/compiled/renderer.js
lineno=282
colno=483
error=TypeError: Cannot read property 'getBoundingClientRect' of null
```

I am a new user, so feel free to correct me if I'm just missing something. I was looking for functionality to navigate via keyboard through recent files, and the option + left or option + right arrow keys were not working for me, so I figured this was the next best thing.

**Screenshots/Demo**

![image](https://user-images.githubusercontent.com/1375869/111043246-6f492900-8439-11eb-82c8-914cd544871e.png)

**Athens Version**

Athens: Version 1.0.0-beta.56 (1.0.0-beta.56)
OS: MacOS Catalina 10.15.7 (19H524)

Using desktop version installed via DMG

**Comment by tangjeff0** [comment]
This is a good bug. Nothing should happen if the user presses up/down.

**Comment by Scatchell** [comment]
Cool! Just to add in case it is related: a similar exception also occurs when I try to tag an item by adding a hash to a word and then immediately hit up/down to try to navigate to the next block.

![image](https://user-images.githubusercontent.com/1375869/111145385-13d58300-8580-11eb-9564-40902f57b70c.png)

**Comment by tangjeff0** [comment]
Good catch. Both of these use the same underlying code, so should be the same fix!

**Comment by hewrin** [comment]
Mind if I take a stab at this?

**Comment by hewrin** [comment]
The problematic functions seem to be `scroll-if-needed` and `is-beyond-rect?` in `athens.util.cljs`

Doing something like this seems to fix it

```
(defn is-beyond-rect?
  "Checks if any part of the element is above or below the container's bounding rect"
  [element container]
  (if (not (nil? element))
      (let [el-box (.. element getBoundingClientRect)
        cont-box (.. container getBoundingClientRect)]
    (or
      (> (.. el-box -bottom) (.. cont-box -bottom))
      (< (.. el-box -top) (.. cont-box -top))))))
 ```
 
But I think there might be a more elegant way. Open to suggestions!

**Comment by phrohdoh** [comment]
> The problematic functions seem to be `scroll-if-needed` and `is-beyond-rect?` in `athens.util.cljs`
> 
> Doing something like this seems to fix it
> 
> ```
> (defn is-beyond-rect?
>   "Checks if any part of the element is above or below the container's bounding rect"
>   [element container]
>   (if (not (nil? element))
>       (let [el-box (.. element getBoundingClientRect)
>         cont-box (.. container getBoundingClientRect)]
>     (or
>       (> (.. el-box -bottom) (.. cont-box -bottom))
>       (< (.. el-box -top) (.. cont-box -top))))))
>  ```
>  
> But I think there might be a more elegant way. Open to suggestions!
> 
> 

Something along these lines is probably a better fix than what is currently proposed in #834 (which fixes only one of these issues and not by fixing the bounds code but by avoiding calling it entirely).

**Comment by hewrin** [comment]
Just FYI, for the # error you'll need to do this to the `scroll-if-needed` function

```
(defn scroll-if-needed
  ;; https://stackoverflow.com/a/45851497
  [element container]
  (if (not (nil? element))
      (let [e-top (.. element -offsetTop)
        e-height (.. element -offsetHeight)
        e-bottom (+ e-top e-height)
        cs-top (.. container -scrollTop)
        c-height (.. container -offsetHeight)
        cs-bottom (+ cs-top c-height)]
    (->> (cond
           (< e-top cs-top)       e-top
           (< cs-bottom e-bottom) (- e-bottom c-height))
         (scroll-top! container)))))

**Comment by tangjeff0** [comment]
Closed by #877
